src/deploy-cromwell-on-azure/Deployer.cs

```diff
@@ -806,26 +804,53 @@ private Task AssignVmAsContributorToStorageAccountAsync(IIdentity managedIdentit
         private Task<IStorageAccount> CreateStorageAccountAsync()
         {
             return Execute(
-                $"Creating Storage Account: {configuration.StorageAccountName}...",
+                $"Finding Existing Storage Account or Creating New Storage Account: {configuration.StorageAccountName}...",
                 async () =>
                 {
-                    var storageAccount = await azureClient.StorageAccounts
-                        .Define(configuration.StorageAccountName)
-                        .WithRegion(configuration.RegionName)
-                        .WithExistingResourceGroup(configuration.ResourceGroupName)
-                        .CreateAsync(cts.Token);
+                    var storageAccount = await TryGetExistingStorageAccountAsync() ?? await CreateNewStorageAccountAsync();
 
                     cts.Token.ThrowIfCancellationRequested();
-
                     var blobClient = await GetBlobClientAsync(storageAccount);
-                    var defaultContainers = new List<string> { WorkflowsContainerName, InputsContainerName, "cromwell-executions", "cromwell-workflow-logs", "outputs", ConfigurationContainerName };
 
+                    var defaultContainers = new List<string> { WorkflowsContainerName, InputsContainerName, "cromwell-executions", "cromwell-workflow-logs", "outputs", ConfigurationContainerName };
                     await Task.WhenAll(defaultContainers.Select(c => blobClient.GetContainerReference(c).CreateIfNotExistsAsync(cts.Token)));
 
                     return storageAccount;
                 });
         }
 
+        private async Task<IStorageAccount> CreateNewStorageAccountAsync()
+        {
+            return await azureClient.StorageAccounts
+                                        .Define(configuration.StorageAccountName)
+                                        .WithRegion(configuration.RegionName)
+                                        .WithExistingResourceGroup(configuration.ResourceGroupName)
+                                        .WithOnlyHttpsTraffic()
+                                        .CreateAsync(cts.Token);
+        }
+
+        private async Task<IStorageAccount> TryGetExistingStorageAccountAsync()
```

Comment:
This can go away, in favor of using the other overload, that takes the name, everywhere. Same for the batch account.

Reply:
I have removed these overloads.